pkg/informer/informer.go

```diff
@@ -104,20 +147,34 @@ func NewDynamicDiscoverySharedInformerFactory(
 	}
 
 	f := &DynamicDiscoverySharedInformerFactory{
-		dynamicClient:     metadataClusterClient,
-		filterFunc:        filterFunc,
-		indexers:          indexers,
-		crdIndexer:        crdInformer.Informer().GetIndexer(),
-		crdInformerSynced: crdInformer.Informer().HasSynced,
+		genericDynamicDiscoverySharedInformerFactory: genericDynamicDiscoverySharedInformerFactory[kcpcache.ScopeableSharedIndexInformer, kcpcache.GenericClusterLister, kcpinformers.GenericClusterInformer]{
+			filterFunc:            filterFunc,
+			indexers:              indexers,
+			listGVRs:              gvrSource.ListGVRs,
```

Comment:
why have function pointers to individual work, as opposed to passing the source?

Reply:
fixed